sdks/java/io/google-cloud-platform/build.gradle

```diff
@@ -50,50 +50,54 @@ dependencies {
     exclude group: 'io.grpc', module: 'grpc-grpclb'
   }
   compile library.java.google_cloud_core
-  compile(library.java.google_cloud_core_grpc) {
-    exclude group: 'io.grpc', module: 'grpc-core' // Use Beam's version
-  }
   compile library.java.google_cloud_datastore_v1_proto_client
   compile library.java.google_cloud_pubsublite
   // GCP PubSub client is used in TestPubSub
   compile library.java.google_cloud_pubsub
   compile library.java.google_cloud_spanner
   compile library.java.google_http_client
   compile library.java.google_http_client_jackson2
-  compile library.java.grpc_alts
   compile library.java.grpc_auth
-  compile library.java.grpc_core
-  compile library.java.grpc_context
-  compile library.java.grpc_grpclb
+  testImplementation 'io.grpc:grpc-core:1.27.2'
   compile library.java.grpc_netty
-  compile library.java.grpc_netty_shaded
-  compile library.java.grpc_stub
   compile library.java.grpc_google_cloud_pubsub_v1
-  compile library.java.grpc_google_cloud_pubsublite_v1
   compile library.java.guava
-  compile library.java.hamcrest_core
   compile library.java.jackson_databind
   compile library.java.joda_time
   compile library.java.junit
   compile library.java.netty_handler
-  compile library.java.netty_tcnative_boringssl_static
-  compile library.java.proto_google_cloud_bigquery_storage_v1beta1
   compile library.java.proto_google_cloud_bigtable_v2
   compile library.java.proto_google_cloud_datastore_v1
   compile library.java.proto_google_cloud_pubsub_v1
   compile library.java.proto_google_cloud_pubsublite_v1
-  compile library.java.proto_google_cloud_spanner_admin_database_v1
   compile library.java.proto_google_common_protos
   compile library.java.protobuf_java
   compile library.java.slf4j_api
+  compile "com.google.api.grpc:proto-google-cloud-bigquerystorage-v1:0.90.0"
```

Comment:
Many of these `com.google` dependencies should read from BeamModulePlugin, which uses the version from `google_cloud_platform_libraries_bom`.

For example: https://github.com/apache/beam/blob/809a0b7b69c335236bb2a4ffea5bbda82416d9ab/buildSrc/src/main/groovy/org/apache/beam/gradle/BeamModulePlugin.groovy#L522

Except we need to add `com.google.api.grpc:proto-google-cloud-bigquerystorage-v1` since it is not exactly the same as `com.google.api.grpc:proto-google-cloud-bigquerystorage`.

Reply:
@ibzib Comparing the lines added here with what's available in BeamModulePlugin and my proposed plan to use what's available where possible:

- com.google.api.grpc:proto-google-cloud-bigquerystorage-v1 is not available - as you pointed out.
- com.google.api.grpc:proto-google-cloud-bigtable-admin-v2 is not available
- com.google.api.grpc:proto-google-cloud-spanner-v1 is not available
- com.google.api:api-common *is available* - but the version coming in from google_cloud_platform_libraries_bom is unclear to me. **Replacing**.
- com.google.apis:google-api-services-storage is a match. **Replacing**.
- com.google.code.gson:gson:2.8.6 is a match. **Replacing**.
- com.google.errorprone:error_prone_annotations is 2.3.3 here, but 2.3.1 in BeamModulePlugin. **Skipping**, but putting in a TODO in the code to match version numbers.
- com.google.protobuf:protobuf-java-util is 3.11.1 here, but 3.12.0 in BeamModulePlugin. **Skipping**, but putting in a TODO in the code to match version numbers.
- io.grpc:grpc-api *is available* - but the version coming in from google_cloud_platform_libraries_bom is unclear to me. **Replacing**.
- org.apache.beam:beam-vendor-guava-26_0-jre is available. ** Replacing**.
- org.apache.httpcomponents:httpclient is 4.5.12 here, but 4.5.10 in BeamModulePlugin. **Skipping** but putting in a TODO in the code to match version numbers.
- org.apache.httpcomponents:httpcore is 4.4.13 here, but 4.4.12 in BeamModulePlugin. ** Skipping** but putting in a TODO in the code to match version numbers.
- org.hamcrest:hamcrest is not available in BeamModulePlugin
- org.threethen:threetenbp is not available in BeamModulePlugin
- compile "org.apache.beam:beam-vendor-grpc-1_26_0:0.3" is available. **Replacing**

Reply:
> the version coming in from google_cloud_platform_libraries_bom is unclear to me.

That's totally fine. The whole point of using the bom is that it provides a single source of truth for dependency versions. If the version provided by the bom mismatches some version the build is actually using, it's a problem.

> org.hamcrest:hamcrest is not available in BeamModulePlugin

I've seen this one in enough other modules that it probably should be placed in BeamModulePlugin.

> Skipping, but putting in a TODO in the code to match version numbers.

Try replacing these anyway, don't worry about matching the versions exactly. In most cases I expect that it shouldn't matter if we use a slightly newer or older version than one of the transitive dependencies, as long as the versions are compatible. (If we don't match versions, Gradle will do that for us anyway when it [resolves dependencies](https://docs.gradle.org/current/userguide/dependency_resolution.html).)

Reply:
OK. Sounds good. I won't bother with the TODOs, and I'll add hamcrest to BeamModulePlugin.

Reply:
Please add the "not available" dependencies to BeamModulePlugin so that we can use the bom versions.